tools/common/StaticRegistrar.cs

```diff
@@ -4945,10 +4945,12 @@ string TryGeneratePInvokeWrapper (PInvokeWrapperGenerator state, MethodDefinitio
 
 			string wrapperName;
 			if (!signatures.TryGetValue (signature.ToString (), out wrapperName)) {
-				var name = "xamarin_pinvoke_wrapper_" + method.Name;
+				var methodName = method.Name.Replace ('<', '_').Replace ('>', '_').Replace ('|', '_');
```

Comment:
I wish we had: 
```csharp
str.ReplaceAll(new [] {'a', 'b'}, '_');
```

Reply:
You can likely use something like `string.Create` to replace those chars instead. Also avoids allocations :D

```csharp
var originalName = method.Name;
var methodName = string.Create (originalName.Length, originalName, static (span, state) => {
    ReadOnlySpan<char> banned = stackalloc[] { '<', '>', '|' };
    for (int i < 0; i < state.Length; ++i) {
        var ch = state[i];
        span[i] = banned.Contains(ch) ? '_' : ch;
    }
});
```